oak-run/src/main/java/org/apache/jackrabbit/oak/index/merge/IndexDefMergerUtils.java

```diff
@@ -163,7 +165,9 @@ private static JsonObject mergeChild(String path, String child, int level, JsonO
         if (level == 0 && USE_PRODUCT_CHILD_LEVEL_0.contains(child)) {
             return p;
         }
-        if (isSameJson(a, p) || isSameJson(c, p)) {
+        if(c == null && p != null){
+            return p;
```

Comment:
I would add a comment here as well, e.g.

    // if customer removed it, then we put it back

Reply:
Fixed